### DISCUSSION TITLE
List Products using Category Handle

### DISCUSSION THREAD

**Discussion by imash96** [medusajs/medusa#4773] [Feature Requests]
Hi,

Can you please add an option to list product using category handle beacuse {frontend_url}/category/:id will not look professional and the endpoint {backend_url}/store/product-categories?handle=men&expand=products is not returning Variants.

Or can you suggest the code to implement this.

**Comment by tsvetann** [comment]
Depending on what exactly you want to achieve you are probably better off using a search engine such as algolia or meilisearch

**Reply by imash96** [reply]
I want to implement a category page which list all the product in that category along with price of the product 

Like if user vist {frontend_url}/category/[handle] I have to display all the product of the category [handle] with price

**Comment by vholik** [comment]
Try to create your own contoller and extend router with endpoint https://your-backend.com/category/:handle

```
 // index.ts
 router.options("/store/category/:handle", cors.default(storeCors));
  router.get(
    "/store/category/:handle",
    cors.default(storeCors),
    wrapHandler(getProductsByHandle)
  );
  
  // product-controllers.ts
  export const getProductByHandle = async (req, res) => {
  const productService = req.scope.resolve("productService") as ProductService;
  const { handle } = req.params;
  const products = await productService.list(
    { handle }, 
    {
      select: defaultAdminProductFields,
      relations: defaultAdminProductRelations
     // Maybe here you want to make pagination etc.
    }
  );
  res.json(products);
};
```
